### ISSUE TITLE
[Bug] Regression: eksctl 0.183.0 fails to create managed nodegroup in AWS China

### ISSUE THREAD

**Issue by artem-nefedov** [eksctl-io/eksctl#7848]
### What happened?

Using eksctl 0.183 and EKS 1.30, it will fail to create managed nodegroup in AWS China due to stack wait timeout.

### How to reproduce it?

Create EKS 1.30 cluster in aws-cn with 1 managed nodegroup.
Nodegroup uses amd64 spot instances with types `["t3.large","t3a.large"]`.

**Logs**

```
2024-06-24 10:10:20 [ℹ]  eksctl version 0.183.0
2024-06-24 10:10:20 [ℹ]  using region cn-north-1
2024-06-24 10:10:20 [ℹ]  setting availability zones to [cn-north-1a cn-north-1b]
2024-06-24 10:10:20 [ℹ]  subnets for cn-north-1a - public:192.168.0.0/19 private:192.168.64.0/19
2024-06-24 10:10:20 [ℹ]  subnets for cn-north-1b - public:192.168.32.0/19 private:192.168.96.0/19
2024-06-24 10:10:20 [ℹ]  nodegroup "managed-ng-1" will use "" [AmazonLinux2023/1.30]
2024-06-24 10:10:20 [ℹ]  using Kubernetes version 1.30
2024-06-24 10:10:20 [ℹ]  creating EKS cluster "redacted" in "cn-north-1" region with managed nodes
2024-06-24 10:10:20 [ℹ]  1 nodegroup (managed-ng-1) was included (based on the include/exclude rules)
2024-06-24 10:10:20 [ℹ]  will create a CloudFormation stack for cluster itself and 0 nodegroup stack(s)
2024-06-24 10:10:20 [ℹ]  will create a CloudFormation stack for cluster itself and 1 managed nodegroup stack(s)
... <abbreviated>
2024-06-24 10:45:55 [ℹ]  waiting for CloudFormation stack "eksctl-redacted-nodegroup-managed-ng-1"
2024-06-24 10:46:28 [ℹ]  waiting for CloudFormation stack "eksctl-redacted-nodegroup-managed-ng-1"
2024-06-24 10:46:28 [!]  1 error(s) occurred and cluster hasn't been created properly, you may wish to check CloudFormation console
2024-06-24 10:46:28 [ℹ]  to cleanup resources, run 'eksctl delete cluster --region=cn-north-1 --name=redacted'
2024-06-24 10:46:28 [✖]  exceeded max wait time for StackCreateComplete waiter
```

**Anything else we need to know?**

Problem is only reproduced in AWS China (global works normally).
Downgrading to eksctl 0.180.0 resolves the problem.

**Versions**

```
eksctl 0.183.0
EKS 1.30
```

**Comment by cPu1** [comment]
@artem-nefedov, can you post your ClusterConfig file or the CLI parameters used to create the cluster, redacting any sensitive information? Can you retry cluster creation after setting `amiFamily` to `AmazonLinux2` and see if that works?

**Comment by artem-nefedov** [comment]
More info.

Managed nodegroup resource in CFN has CreationFailed with this message:

```
Resource handler returned message: "[Issue(Code=NodeCreationFailure, Message=Unhealthy nodes in the kubernetes cluster, ResourceIds=[i-...., i-....])] (Service: null, Status Code: 0, Request ID: null)" (RequestToken: ...., HandlerErrorCode: GeneralServiceException)
```

Cluster config:

```yaml
addons:
- attachPolicyARNs:
  - arn:aws:iam::aws:policy/AmazonEKS_CNI_Policy
  name: vpc-cni
  version: latest
- name: coredns
  version: latest
- name: kube-proxy
  version: latest
apiVersion: eksctl.io/v1alpha5
cloudWatch:
  clusterLogging:
    enableTypes:
    - audit
    - authenticator
    - scheduler
iam:
  serviceAccounts:
  # redacted
  withOIDC: true
kind: ClusterConfig
managedNodeGroups:
- desiredCapacity: 2
  instanceTypes:
  - t3.large
  - t3a.large
  maxSize: 10
  minSize: 2
  name: managed-ng-1
  privateNetworking: true
  spot: true
  tags:
   # redacted
  volumeEncrypted: true
  volumeSize: 20
  volumeType: gp3
metadata:
  name: # redacted
  region: cn-north-1
  tags:
    # redacted
  version: "1.30"
secretsEncryption:
  keyARN: # redacted

```

**Comment by artem-nefedov** [comment]
@cPu1 eksctl 0.180.0 successfully creates `AmazonLinux2023` nodegroup, so ami family should be fine

**Comment by artem-nefedov** [comment]
It looks like this is fixed with eksctl version 0.184.0 (even though there's no mention of it).
The issue is no longer reproduced, so I'm closing this.
